### DISCUSSION TITLE
How to sent a real null in a kafka message?

### DISCUSSION THREAD

**Discussion by poorSock** [sngular/kloadgen#353] [Q&A] [answered]
Hi,
another question from me, maybe someone has a quick hint, what I am doing wrong.

I try to send a real null value in a kafka message (not a "null" String).
In our schema the field is defined as follows:
`{
    "name" : "idReferenceType",
    "type" : [ "null", "string" ],
    "default" : null
  }`

According to the [docs](https://github.com/sngular/kloadgen/wiki/schemas#null-values) sending null should work for this type of field.

When trying to sent an event I only manage to get a "null" String in my kafka message, not a real null value.

I've tried:
![image](https://user-images.githubusercontent.com/84520839/225328003-46c11ba1-8beb-4175-a340-3f62db314a61.png)
![image](https://user-images.githubusercontent.com/84520839/225328084-15f5864a-5bef-4f9f-83f0-75987796d70d.png)
![image](https://user-images.githubusercontent.com/84520839/225328175-2e31dd44-cef6-4e4b-b5ed-442c94652458.png)
![image](https://user-images.githubusercontent.com/84520839/225328340-d151bf9c-be54-430b-b65d-344544ba2435.png)
![image](https://user-images.githubusercontent.com/84520839/225328511-015f583c-771d-4f8a-a051-46822ef09786.png)

But all of these only lead to a "null" string or empty "" string instead of a real null value.

How to sent a real null?

tested with V 5.3.1 & V5.3.3

**Comment by marioTemprano** [comment, marked as answer]
Hi @poorSock, if you remove the "idReferenceType" line, it will send a real null value

**Comment by poorSock** [comment]
Hi,
thanks, that does it. I just wonder, what is the use of the "null" type then?

**Comment by marioTemprano** [comment]
@poorSock, the "null" type is mainly for union definition
